### PR TITLE
chore: fix typings and expose asyncIterator 

### PR DESCRIPTION
**Pull request by menduz**

### CODE REVIEW

File: src/ContentClient.ts [hunk 1]
```diff
@@ -332,8 +343,10 @@ export class ContentClient implements ContentAPI {
         } catch (error) {
           if (errorListener) {
             errorListener(`${error}`)
+            exit = true
+          } else {
+            throw error
```
Comment (menduz) [inline src/ContentClient.ts:348]: 🌷

File: src/ContentAPI.ts [hunk 2]
```diff
@@ -33,17 +38,25 @@ export interface ContentAPI {
...
   downloadContent(contentHash: ContentFileHash, options?: RequestOptions): Promise<Buffer>
   isContentAvailable(cids: ContentFileHash[], options?: RequestOptions): Promise<AvailableContentResult>
-  pipeContent(
-    contentHash: ContentFileHash,
-    writeTo: ReadableStream<Uint8Array>,
-    options?: RequestOptions
-  ): Promise<Map<string, string>>
+
+  /**
+   * pipeContent only works in Node.js like environments
```
Comment (agusaldasoro) [inline src/ContentAPI.ts:57]: should we deprecate this too?

File: src/ContentClient.ts [hunk 3]
```diff
@@ -106,9 +105,7 @@ export class ContentClient implements ContentAPI {
     // Check if we are running in node or browser
     const areWeRunningInNode = isNode()
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    const form: FormData = areWeRunningInNode ? new NodeFormData() : new FormData()
+    const form = new FormData()
```
Comment (agusaldasoro) [inline src/ContentClient.ts:108]: how is it that this works now?

File: src/ContentClient.ts [hunk 4]
```diff
@@ -403,7 +420,14 @@ export type DeploymentOptions<T> = {
...
   sortBy?: DeploymentSorting
   fields?: DeploymentFields<T>
+  /**
+   * @deprecated please use try-catch and iterators instead
+   */
   errorListener?: (errorMessage: string) => void
+  /**
+   * Amount of elements per page
+   */
+  limit?: number
```
Comment (agusaldasoro) [inline src/ContentClient.ts:430]: isn't `limit` already in `DeploymentFilters`? If not, it should be there (in catalyst-commons)

File: src/LambdasClient.ts [hunk 5]
```diff
@@ -92,15 +92,15 @@ export class LambdasClient implements LambdasAPI {
   }
 
   fetchCatalystsApprovedByDAO(options?: RequestOptions): Promise<ServerMetadata[]> {
-    return this.fetcher.fetchJson(`${this.lambdasUrl}/contracts/servers`, options)
+    return this.fetcher.fetchJson(`${this.lambdasUrl}/contracts/servers`, options) as any
```
Comment (agusaldasoro) [inline src/LambdasClient.ts:95]: why is `as any` needed?

File: src/ContentClient.ts (hunk 3 above)
Comment (menduz) [reply src/ContentClient.ts:108]: FormData works both for browsers and node.js https://github.com/form-data/form-data/blob/master/lib/browser.js

File: src/ContentAPI.ts (hunk 2 above)
Comment (menduz) [reply src/ContentAPI.ts:57]: I prefer to do it in a different PR to not block the fix of this PR

**Review by agusaldasoro — APPROVED** [review]

**Pull request merged**